demo/src/main.rs

```diff
@@ -21,6 +21,9 @@ include_cpp! {
     generate!("Goat")
 }
 
+#[link(name = "autocxx-demo")]
+extern "C" {}
+
```

Comment:
Why this change?

Reply:
That's how the .rs files generated by integration-tests get the static library linked in, and it seemed like a reasonable thing to add for a complete demo that can be built either way the docs mention.

Reply:
Ah right, so this is on the basis that the demo code should build using `autocxx-gen` as well as `autocxx-build`?

I'd prefer to keep the demo code as simple as possible for the Cargo (`autocxx-build`) world, where this is already taken care of by the `build.rs`. I think anyone deviating from Cargo knows that they will need to jump through some hoops, and if they're using blaze/bazel/gn or similar, this `link` directive probably won't do the trick anyway.

Reply:
Yes, exactly.

In that case, any thoughts on adding something for `autocxx-gen` tests to build? Could copy the whole directory, but that seems annoying.

Also, I'll back the tests out of this PR for now in the interests of getting the functional change merged.

Reply:
Ah right, it's used by the test code, that's the bit I hadn't realized.

Yeah - I'd still strongly prefer to avoid adding these lines into the demo. I want that to be focused on the simplest possible experience for cargo users. I'd be OK with adding another copy of equivalent files for the `autocxx_gen` tests, or programmatically writing files out from the test code.